clients/privacy-center/common/location.ts

```diff
@@ -0,0 +1,75 @@
+import type { NextApiRequest } from "next";
+
+// DEFER: Import this type from fides-js when it exists!
+// (see https://github.com/ethyca/fides/pull/3191)
+// import type { UserGeolocation } from "fides-js";
+export type UserGeolocation = {
+  country?: string; // "US"
+  ip?: string; // "192.168.0.1:12345"
+  location?: string; // "US-NY"
+  region?: string; // "NY"
+};
+
+// Regex to validate a location string, which must:
+// 1) Start with a 2-3 character country code (e.g. "US")
+// 2) Optionally end with a 2-3 character region code (e.g. "CA")
+// 3) Separated by a dash (e.g. "US-CA")
+const VALID_ISO_3166_LOCATION_REGEX = /^\w{2,3}(-\w{2,3})?$/;
+
+// Constants for the supported CloudFront geolocation headers
+// (see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/adding-cloudfront-headers.html#cloudfront-headers-viewer-location)
+const CLOUDFRONT_HEADER_COUNTRY = "cloudfront-viewer-country";
+const CLOUDFRONT_HEADER_REGION = "cloudfront-viewer-country-region";
+export const LOCATION_HEADERS = [
+  CLOUDFRONT_HEADER_COUNTRY,
+  CLOUDFRONT_HEADER_REGION,
+];
+
+/**
+ * Lookup the "location" (ie country and region) for the given request by looking for either:
+ * 1) Supported geolocation headers (e.g. "Cloudfront-Viewer-Country: US")
+ * 2) An explicit "location" query param (e.g. https://privacy.example.com/some/path?location=US-CA)
```

Comment:
nit: I feel like these should be swapped in order since we prioritize the location param over the geolocation header

Reply:
Good call, fixed!